src/lsm/manifest_log.zig

```diff
@@ -282,11 +287,8 @@ pub fn ManifestLogType(comptime Storage: type, comptime TableInfo: type) type {
             assert(table.snapshot_min > 0);
             assert(table.snapshot_max > table.snapshot_min);
 
-            if (manifest_log.blocks.empty()) {
-                manifest_log.acquire_block();
-            } else if (manifest_log.entry_count == entry_count_max) {
-                assert(manifest_log.blocks.count > 0);
-                manifest_log.close_block();
+            if (manifest_log.entry_count == 0) {
+                assert(manifest_log.blocks.count == manifest_log.blocks_closed);
```

Comment:
Much nicer like this:
* Both conditions are all in terms of `entry_count` now.
* We collapse two branches into one.
* We no longer have a TOCTOU between appending to a block and closing a block.

Reply:
I'm puzzled why I did it the other way! 😁

Reply:
Ah, I think it was because the append might not happen at the same time as when we flush the block. So the other way we can handle 1 more insert, without needing the blocks ring buffer to have a whole new completely empty block allocated.

But this way is still better.